packages/react-vapor/src/components/flatSelect/FlatSelect.tsx

```diff
@@ -15,62 +21,68 @@ export interface IFlatSelectOwnProps {
     classes?: string[] /* @deprecated use className instead */;
 }
 
-export interface IFlatSelectStateProps {
-    selectedOptionId?: string;
-}
-
-export interface IFlatSelectDispatchProps {
-    onRender?: () => void;
-    onDestroy?: () => void;
-    onOptionClick?: (option: IFlatSelectOptionProps) => void;
-}
+export type IFlatSelectProps = IFlatSelectOwnProps &
+    ReturnType<typeof mapStateToProps> &
+    ReturnType<typeof mapDispatchToProps>;
 
-export interface IFlatSelectProps extends IFlatSelectOwnProps, IFlatSelectStateProps, IFlatSelectDispatchProps {}
+export const FlatSelect: React.FunctionComponent<IFlatSelectProps> = ({
+    onDestroy,
+    onRender,
+    onOptionClick,
+    onClick,
+    options,
+    disabled,
+    selectedOptionId,
+    group,
+    optionPicker,
+    classes,
+    className,
+}) => {
+    React.useEffect(() => {
+        onRender?.();
+        return onDestroy;
+    }, []);
 
-export class FlatSelect extends React.Component<IFlatSelectProps> {
-    componentWillMount() {
-        if (this.props.onRender) {
-            this.props.onRender();
-        }
-    }
+    const handleOnOptionClick = (option: IFlatSelectOptionProps) => {
+        onOptionClick?.(option);
+        onClick?.(option);
+    };

```

Comment:
its only a details "de téteux" but since this handles the `onOptionClick` as well as the `onClick` this may could be renamed `handleClick`